test/cpp/interop/xds_interop_client.cc

```diff
@@ -265,20 +269,22 @@ class LoadBalancerStatsServiceImpl : public LoadBalancerStatsService::Service {
                         LoadBalancerStatsResponse* response) override {
     int start_id;
     int end_id;
-    XdsStatsWatcher* watcher;
+    std::unique_ptr<XdsStatsWatcher> watcher;
     {
       std::lock_guard<std::mutex> lock(stats_watchers_->mu);
       start_id = stats_watchers_->global_request_id + 1;
       end_id = start_id + request->num_rpcs();
-      watcher = new XdsStatsWatcher(start_id, end_id);
-      stats_watchers_->watchers.insert(watcher);
+      watcher = std::make_unique<XdsStatsWatcher>(
+          start_id, end_id,
+          std::vector<std::string>(request->metadata_keys().begin(),
+                                   request->metadata_keys().end()));
+      stats_watchers_->watchers.insert(watcher.get());
     }
-    watcher->WaitForRpcStatsResponse(response, request->timeout_sec());
+    *response = watcher->WaitForRpcStatsResponse(request->timeout_sec());
```

Comment:
Why was the `response` pointer changed to returned struct? Isn't this an unnecessary copy of the object?

Reply:
Just a cleaner API. No need to allocate object outside, no risk of `null` being passed by mistake, etc. Compiler will likely do copy elision so there will be no performance impact.

Reply:
> ...No need to allocate object outside

Well, if you are doing

```
*response = ...
```

You are assuming `response` is pointing to an already allocated object. In the previous code the same pointer was being passed inside and now the returned value (an object) is being assigned to `*response`. But I agree compiler will (might?) do copy elision so perf impact might be zero.